File: backend/app/modules/reading_plan/schemas/book.py
from datetime import date
from typing import Optional

from app.modules.reading_plan.schemas.base import CamelModel


class BookSearchResult(CamelModel):
    isbn13: str
    title: str
    author: Optional[str] = None
    publisher: Optional[str] = None
    cover_url: Optional[str] = None
    page_count: Optional[int] = None
    published_date: Optional[date] = None
    description: Optional[str] = None


class BookSearchResponse(CamelModel):
    items: list[BookSearchResult]
    total_count: int

File: frontend/src/types/reading-plan/book.ts
export type LibraryStatus = "WISH" | "READING" | "COMPLETED";

export interface Book {
  id: string;
  isbn13: string;
  title: string;
  author: string;
  publisher: string;
  coverUrl?: string | null;
  pageCount?: number | null;
  description?: string | null;
  publishedDate?: string | null;
}

export interface BookSearchResult {
  isbn13: string;
  title: string;
  author: string | null;
  publisher: string | null;
  coverUrl: string | null;
  pageCount: number | null;
  publishedDate: string | null;
  description: string | null;
}

export interface UserLibraryItem {
  id: string;
  book: Book;
  status: LibraryStatus;
  currentPage: number;
  rating?: number | null;
  startedAt?: string | null;
  completedAt?: string | null;
}

export interface MonthlyGoal {
  yearMonth: string;
  target: number | null;
  completed: number;
}

export interface ProgressLogEntry {
  id: string;
  libraryId: string;
  page: number | null;
  percent: number | null;
  memo: string | null;
  recordedAt: string;
}

export interface LibraryComment {
  id: string;
  libraryId: string;
  book: {
    id: string;
    title: string;
    coverUrl?: string | null;
  };
  memo: string;
  recordedAt: string;
}
